src/client/src/routes/NotificationRoute.tsx

```diff
@@ -36,19 +36,16 @@ export class NotificationRoute extends React.Component<{}, State> {
     fin.desktop.InterApplicationBus.publish(InteropTopics.HighlightBlotter, this.state.message)
 
   render() {
-    let { message } = this.state
-    if (!message) {
-      message = { tradeNotification: undefined }
-    }
-    return (
+    const { message } = this.state
+    return !!message ? (
       <ThemeProvider>
         <TradeNotification
           trade={message.tradeNotification}
           dismissNotification={this.onDismissNotification}
           highlightTradeInBlotter={this.highlightTradeInBlotter}
         />
       </ThemeProvider>
-    )
+    ) : null
```

Comment:
Wouldn't this previously always render `<TradeNotification />` but `message.tradeNotification` would be undefined? Or was that actually an erroneous scenario?

Reply:
Ah I see now. The `TradeNotification` without a `trade` (or with the `defaultProps.trade` is.... Odd.

Reply:
Yeah, doesn't make sense to have a notification when all the details are not present.